projects/components/src/select/select.component.ts

```diff
@@ -55,7 +55,13 @@ import { SelectSize } from './select-size';
               class="trigger-content menu-with-border"
               [ngClass]="[this.justifyClass]"
             >
-              <ht-icon *ngIf="this.icon" class="trigger-prefix-icon" [icon]="this.icon" [size]="this.iconSize">
+              <ht-icon
+                *ngIf="this.getPrefixIcon | htMemoize: selected:this.icon"
```

Comment:
simple conditional logic inside probably doesn't need a memoize (also, this.icon is accessible inside the function, anyway). Reserve the confusing syntax for heavy logic or anywhere that returns a new reference each time.

Reply:
fixed.